### ISSUE TITLE
stun server with tcp

### ISSUE THREAD

**Issue by H777K** [netbirdio/netbird#2032]
**Describe the problem**

Hi i am currently setting up a netbird self-hosted setup. I use coturn as stun/turn server. Coturn is exposed with port 80/tcp and internally forwarded to port 3478/tcp. In the management.json I have specified the tcp protocol for both the stun and the turn server. The turn server works perfectly, but the stun server does not:

**netbird status -d output:**

```
Relays: 
  [stun:<dns_name_placeholder>:80] is Unavailable, reason: stun request: context deadline exceeded
  [turn:<dns_name_placeholder>:80?transport=tcp] is Available
```

**content of management.json:**

```json
{
    "Stuns": [
	{
            "Proto": "tcp",
            "URI": "stun:<dns_name_placeholder>:80",
            "Username": "",
            "Password": ""
        }
    ],
    "TURNConfig": {
        "TimeBasedCredentials": false,
        "CredentialsTTL": "0s",
        "Secret": "",
        "Turns": [
            {
                "Proto": "tcp",
                "URI": "turn:<dns_name_placeholder>:80?transport=tcp",
                "Username": "myuser",
                "Password": "mypassword"
            },
        ]
    },
```

If i set `stun:<dns_name_placeholder>:80?transport=tcp` in the management.json, the stun server no longer appears when i execute `netbird status -d`. If i remove the stun server completely from the management.json, everything still works.
This leads to some questions for me:
1. How can i use the stun server with tcp, is something missing in my configuration?
2. Is a stun server mandatory or is it sufficient to use just a turn server?
3. What are the disadvantages of using the stun/turn server with tcp instead of udp?

**Expected behavior**

I would expect the stun server to also work via tcp, like the turn server

**Are you using NetBird Cloud?**

no (self-hosted NetBird's control plane)

**NetBird version**

0.27.7

**Comment by H777K** [comment]
@lixmal Thank you for the explanation

**Comment by Mark3xtrm** [comment]
I'm encountering this as well. Same error. I think I've opened all the right ports on my router (which is behind a modem that gives it the wan connection), I wrote firewall traffic rules (OpenWrt) but nothing to do, I can't reach out my stun/turn server (which is on a router vlan, btw). Idk if it's useful, but I'm using Traefik.

So my peers in lan can see each other and can ping themselves at the mesh IPs (after I wrote some traffic rules for the communication between their vlans), but my 4G (wan) Android peer can't even ping the ones in the lan - of course it's inside the vpn mesh and I can see it connected to it while it's in 4G.

On [Trickle ICE](https://webrtc.github.io/samples/src/content/peerconnection/trickle-ice/) I always end up like this
![image](https://github.com/user-attachments/assets/865129d0-c9ba-4582-ac04-7c03130ccff4)
